leaderelection/src/test/java/io/scalecube/leaderelection/LeaderElectionIT.java

```diff
@@ -0,0 +1,118 @@
+package io.scalecube.leaderelection;
+
+import static org.junit.Assert.assertTrue;
+
+import io.scalecube.cluster.Cluster;
+import io.scalecube.cluster.ICluster;
+import io.scalecube.transport.Address;
+
+import com.google.common.base.Throwables;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class LeaderElectionIT {
+
+  Executor exec = Executors.newCachedThreadPool();
+  @Test
+  public void testAllNodesFollowOneLeader() {
+    ICluster seed = Cluster.joinAwait();
+    LeaderElection seedLeader = createLeaderElection(seed);
+    List<LeaderElection> leaders = createCluster(seed);
+
+    // waiting for a leader to be selected
+    awaitSeconds(30);
+
+    Address seedLeaderAddress = seedLeader.leader();
+    
+    for (LeaderElection m : leaders) {
+      Address memberLeaderAddress = m.leader();
+      assertTrue("Expected leader address " + seedLeaderAddress + ", but actual: " + memberLeaderAddress,
+          memberLeaderAddress.equals(seedLeaderAddress));
+    }
+    
+    // shutdown everything
+    for (LeaderElection m : leaders) {
+      m.cluster().shutdown();
+    }
+    seed.shutdown();
+  }
+
+  
+  @Test
+  public void testLeaderDiesAndNewLeaderIsElected() {
+    ICluster seed = Cluster.joinAwait();
+    LeaderElection seedLeader = createLeaderElection(seed);
+    List<LeaderElection> leaders = createCluster(seed);
+
+    // waiting for a leader to be selected
+    awaitSeconds(30);
```

Comment:
there is random timeout to select a leader the default is min 15 max 30 seconds (with some random factor)
- when timeout exceeds on first node it start selection
when next node comes leader welcome him specifically so time for his is intimidate
when node dies there are already random clocks waiting for the heartbeats so time should be much less